test/core/block.wast

```diff
@@ -1470,6 +1470,12 @@
   ))
   "type mismatch"
 )
+(assert_invalid
+  (module (func $type-param-nested-poly-num-nondrop
+    (unreachable) (block (param i32))
```

Comment:
I don't think that this is checking what you think. The type error here has nothing to do with the unreachable (which is fine before a param block). The error is the block not dropping its param, so that its body doesn't have type [i32] -> [], as it should according to the type annotation.

Reply:
The problem was that particular implementation was using common stack for storing types during validation and the validator incorrectly assumed polymorphic stack state for inner block (due to its params).

This test checks the above.

Reply:
To clarify, there are two possible tests you may care about:
```
(func (block (param i32)))
```
This should fail for the same reason your original test did. The unreachable was immaterial.

Or:
```
(func (unreachable) (block (param i32) (drop)))
```
This should succeed.